cmd/kubeapps-apis/server/packages.go

```diff
@@ -68,3 +68,41 @@ func (s packagesServer) GetAvailablePackageSummaries(ctx context.Context, reques
 		AvailablePackagesSummaries: pkgs,
 	}, nil
 }
+
+// GetAvailablePackages returns the packages based on the request.
+func (s packagesServer) GetAvailablePackageDetail(ctx context.Context, request *packages.GetAvailablePackageDetailRequest) (*packages.GetAvailablePackageDetailResponse, error) {
+	contextMsg := ""
+	if request.AvailablePackageRef != nil && request.AvailablePackageRef.Context != nil {
+		contextMsg = fmt.Sprintf("(cluster=[%s], namespace=[%s])", request.AvailablePackageRef.Context.Cluster, request.AvailablePackageRef.Context.Namespace)
+	}
+
+	log.Infof("+core GetAvailablePackageDetail %s", contextMsg)
+
+	pkg := &packages.AvailablePackageDetail{}
+
+	// TODO: We can do these in parallel in separate go routines.
+	for _, p := range s.plugins {
```

Comment:
For this call (singular - GetAvailablePackageDetail) it doesn't make sense to iterate the plugins, as it only makes sense to call the particular plugin identified in the request context? The request must include the plugin in the context and only that plugin should be used.